cpp/ql/lib/semmle/code/cpp/exprs/Access.qll

```diff
@@ -314,7 +314,7 @@ private predicate exprHasReferenceConversion(Expr e) { referenceConversion(e.get
 class ImplicitThisFieldAccess extends FieldAccess {
   override string getAPrimaryQlClass() { result = "ImplicitThisFieldAccess" }
 
-  ImplicitThisFieldAccess() { not exists(this.getQualifier()) }
+  ImplicitThisFieldAccess() { this.getQualifier().isCompilerGenerated() or not exists(this.getQualifier()) }
```

Comment:
I think the "note" in the QLdoc of this predicate can now be removed?

Reply:
I now wonder if we should be more specific and check that the qualifier is actually `this` (if it exists and is generated)?